src/pkcs15init/pkcs15-oberthur-awp.c

```diff
@@ -967,7 +967,7 @@ awp_encode_cert_info(struct sc_pkcs15_card *p15card, struct sc_pkcs15_object *ob
 	 */
 	ptr = awp_get_commonName(x);
 	if (!ptr)
-		LOG_TEST_RET(ctx, SC_ERROR_INTERNAL, "AWP encode cert failed: cannot get CommonName");
+		LOG_TEST_GOTO_ERR(ctx, SC_ERROR_INTERNAL, "AWP encode cert failed: cannot get CommonName");
```

Comment:
If I read the macro `LOG_TEST_GOTO_ERR` correctly, it does not set the `r` variable, which is used as a return value of the function. This means that the function is going to return the previous value of `r`, which is in this case `0`, even on errors from these conditions, that are not modifying `r`.